src/kOS.Safe/Encapsulation/EnumerableValue.cs

```diff
@@ -1,26 +1,30 @@
-﻿using System.Collections;
+﻿using kOS.Safe.Encapsulation.Suffixes;
+using kOS.Safe.Serialization;
+using System.Collections;
 using System.Collections.Generic;
 using System.Linq;
-using kOS.Safe.Encapsulation.Suffixes;
-using kOS.Safe.Serialization;
 
 namespace kOS.Safe.Encapsulation
 {
     [kOS.Safe.Utilities.KOSNomenclature("Enumerable")]
-    public abstract class EnumerableValue<T, TC> : SerializableStructure, IEnumerable<T> where TC : IEnumerable<T> where T : Structure
+    public abstract class EnumerableValue<T, TE> : SerializableStructure, IEnumerable<T>
+        where TE : IEnumerable<T>
+        where T : Structure
     {
-        protected TC Collection { get; private set; }
+        protected TE InnerEnumerable { get; private set; }
+        private readonly string label;
 
-        protected EnumerableValue(TC collection)
+        protected EnumerableValue(string label, TE enumerable)
         {
-            Collection = collection;
+            this.label = label;
```

Comment:
Yeah, I couldn't make those types of changes in the initial review of Tom's PRs because I didn't have Nomenclature.  I intend to go through and find this kind of thing if possible.  I also want to make Argument Mismatch exceptions show the new Nomenclature names.